app/models/alchemy/page/page_elements.rb

```diff
@@ -143,6 +143,16 @@ def element_definitions
       @_element_definitions ||= element_definitions_by_name(element_definition_names)
     end
 
+    # All element definitions defined for page's page layout including nestable element definitions
+    #
+    def descendent_element_definitions
+      definitions = element_definitions_by_name(element_definition_names)
+      definitions.select { |d| d.key?('nestable_elements') }.each do |d|
+        definitions += element_definitions_by_name(d['nestable_elements'])
+      end
+      definitions
```

Comment:
@robinboening no, this could happen, consider following:

```ruby
[
  {'name' => 'slider', 'nestable_elements' => %w(slide)},
  {'name' => 'gallery', 'nestable_elements' => %w(slide)},
  {'name' => 'slide'}
]
```

directly from my amended test case in b6d70f4 ;)